OneDrivePhotoBrowser/AccountSelection.xaml.cs

```diff
@@ -51,28 +64,33 @@ public AccountSelection()
 
         private async void AccountSelection_Loaded(object sender, RoutedEventArgs e)
         {
-            if (((App)Application.Current).OneDriveClient != null)
+            var app = ((App) Application.Current);
+            if (app.OneDriveClient != null)
             {
-                await ((App)Application.Current).OneDriveClient.SignOutAsync();
-
-                var client = ((App)Application.Current).OneDriveClient as OneDriveClient;
-                if (client != null)
+                var msaAuthProvider = app.AuthProvider as MsaAuthenticationProvider;
+                var adalAuthProvider = app.AuthProvider as AdalAuthenticationProvider;
```

Comment:
We should fix this, otherwise an Authentication Provider interface has essentially no value.

---
In reply to: [73741840](https://github.com/OneDrive/onedrive-sample-photobrowser-uwp/pull/1#discussion_r73741840) [](ancestors = 73741840)